packages/peregrine/lib/talons/CategoryTree/useCategoryTree.js

```diff
@@ -41,8 +43,8 @@ export const useCategoryTree = props => {
         }
     }, [data, updateCategories]);
 
-    const rootCategory = categories[categoryId];
-    const { children } = rootCategory || {};
+    const rootCategory = data && data.category;
```

Comment:
The catalog data is still necessary even after this change as it controls the rendering near the top of this tree in `useNavigation`. All that @eug123 did here is change where the category tree (this component) gets the data - instead of using the prop data it just uses the query data that it was making anyways. Functionally identical without the round trip of updating the store and then re-rendering when the `catalog` prop makes its way back here.

I do think we should try to disentangle some of the catalog reducer logic from the navigation drawer but I think that scope is beyond this PR.

Reply:
Other components still use `categories` from Redux. Seems removing `updateCategories()` from here, will require rebuild all of them to use Apollo cache, and eventually deprecate `useCatalogContext`